src/picongpu/include/particles/bremsstrahlung/PhotonEmissionAngle.hpp

```diff
@@ -0,0 +1,264 @@
+{
+namespace bremsstrahlung
+{
+
+namespace detail
+{
+
+/** Functor mapping `delta` to the photon emission polar angle `theta`,
+ * where delta is a uniformly distributed random number between zero and one.
+ */
+struct GetPhotonAngleFunctor
+{
+    typedef typename ::PMacc::result_of::Functor<
+        ::PMacc::cursor::tools::LinearInterp<float_X>,
+        ::PMacc::cursor::BufferCursor<float_X, DIM2> >::type LinInterpCursor;
+
+    typedef float_X type;
+
+    LinInterpCursor linInterpCursor;
+    float_X lnMinGamma;
+    float_X lnMaxGamma;
+
+    /** constructor
+     *
+     * @param linInterpCursor lookup table for the photon emission angle.
+     */
+    HDINLINE GetPhotonAngleFunctor(LinInterpCursor linInterpCursor)
+        : linInterpCursor(linInterpCursor)
+    {
+        this->lnMinGamma = algorithms::math::log(photon::MIN_GAMMA);
+        this->lnMaxGamma = algorithms::math::log(photon::MAX_GAMMA);
+    }
+
+    /** Return the polar emission angle of the photon.
+     *
+     * @param delta uniformly distributed random number between zero and one.
+     * @param gamma relativistic factor of the incident electron.
+     */
+    HDINLINE float_X operator()(const float_X delta, const float_X gamma) const
+    {
+        const float_X binDelta = delta * static_cast<float_64>(photon::NUM_SAMPLES_DELTA - 1);
+
+        const float_X lnGamma = algorithms::math::log(gamma);
+        const float_X binGamma =
+            (lnGamma - this->lnMinGamma) /
+            (this->lnMaxGamma - this->lnMinGamma) *
+            static_cast<float_X>(photon::NUM_SAMPLES_GAMMA - 1);
+
+        if (picLog::log_level & picLog::CRITICAL::lvl)
+        {
+            if(binGamma > float_X(photon::NUM_SAMPLES_GAMMA - 1))
+            {
+                printf("[Bremsstrahlung] error lookup table: gamma = %g is out of range.\n",
+                       gamma);
+            }
+        }
+
+        return this->linInterpCursor[float2_X(binDelta, binGamma)];
+    }
+};
+
+} // namespace detail
+
+/** Creates and holds the lookup table for the photon emission angle.
+ */
+struct GetPhotonAngle
+{
+    typedef detail::GetPhotonAngleFunctor GetPhotonAngleFunctor;
+
+private:
+
+    typedef boost::shared_ptr<PMacc::container::DeviceBuffer<float_X, DIM2> > MyBuf;
+    MyBuf dBufTheta;
+
+    /** probability density at polar angle theta.
+     * It's the ultrarelativistic limit of the dipole radiation formula, see e.g. Jackson, chap. 15.2
+     */
+    struct Probability
+    {
+        const float_64 gamma2;
+        Probability(const float_64 gamma) : gamma2(gamma*gamma) {}
+
+        template<typename T_State>
+        void operator()(const T_State &p, T_State &dpdtheta, const float_64 theta) const
+        {
+            const float_64 theta2 = theta*theta;
+            const float_64 denom = 1.0 + gamma2 * theta2;
+
+            dpdtheta[0] = 3.0 * theta * gamma2 * (1.0 + gamma2*gamma2 * theta2*theta2) /
+                          (denom*denom*denom*denom);
```

Comment:
Here the angular probability is depending on the polar angle `theta` instead of the solid angle `Omega`, so:
```
dp / dtheta = dp / dOmega * 2pi * theta
```